src/components/Table/components/TableFeatures/RowSorting/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { Column, ColumnSort, RowData, Table } from "@tanstack/react-table";
+import { MouseEvent } from "react";
+
+/**
+ * Constructs a `ColumnSort` object from a column.
+ * @param column - Column.
+ * @returns `ColumnSort` object with sorting direction and column ID.
+ */
+export function buildColumnSort<T extends RowData>(
+  column: Column<T>
+): ColumnSort {
+  return { desc: column.getFirstSortDir() === "desc", id: column.id };
+}
+
+/**
+ * Toggles the sorting state for the specified column.
+ * The sorting state of a table is modified based on user interaction, table grouping state, and
+ * its sorting configuration (single-sort or multi-sort modes).
+ * ### Sorting State:
+ * - **Ungrouped Table**:
+ *  - Sorting is toggled normally, respecting single-sort or multi-sort modes.
+ * - **Grouped Table**:
+ *   - **Multi-Sort Request**:
+ *     - Sorting is toggled normally.
+ *   - **Single-Sort Request**:
+ *     - **Multi-Sort Mode**:
+ *       - Grouped column is preserved as the first sort key, and the requested column is sorted as the second sort key.
+ *     - **Single-Sort Mode**:
+ *       - If the grouped column is already sorted, no action taken (the grouped column remains the primary sort key).
+ *       - Otherwise, sorting is toggled normally.
+ * @param mouseEvent - Mouse event.
+ * @param table - Table.
+ * @param column - Column.
+ */
+export function handleToggleSorting<T extends RowData>(
+  mouseEvent: MouseEvent,
+  table: Table<T>,
+  column: Column<T>
+): void {
+  const {
+    getColumn,
+    getState,
+    options: { enableMultiSort, isMultiSortEvent },
+    setSorting,
+  } = table;
+  const { getCanSort, getSortIndex, toggleSorting } = column;
+  const { grouping, sorting } = getState();
+
+  // Table and column sorting is not enabled.
+  if (!getCanSort()) return;
+
+  // Sorting is enabled.
+  // Table is not grouped; toggle sorting as usual.
+  if (grouping.length === 0) {
+    toggleSorting(undefined, isMultiSortEvent?.(mouseEvent));
+    return;
+  }
+
+  // Table is grouped.
+  // Multi-sort mode and multi-sort requested; toggle sorting as usual.
+  if (enableMultiSort && isMultiSortEvent?.(mouseEvent)) {
+    toggleSorting(undefined, true);
```

Comment:
Table is grouped, but user has requested multi-sort; sort as normal (the "grouped" column is either in first sort position, or if is not sortable, ignored anyway).